core/src/main/java/cz/o2/proxima/storage/watermark/GlobalWatermarkThroughputLimiter.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.storage.watermark;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import cz.o2.proxima.storage.ThroughputLimiter;
+import cz.o2.proxima.util.Classpath;
+import cz.o2.proxima.util.ExceptionUtils;
+import cz.o2.proxima.util.Pair;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A {@link ThroughputLimiter} that synchronizes progress of global watermark among distributed
+ * consumers.
+ */
+@Slf4j
+public class GlobalWatermarkThroughputLimiter implements ThroughputLimiter {
+
+  private static final long serialVersionUID = 1L;
+
+  static final String TRACKER_CFG_PREFIX = "tracker.";
+  static final String KW_CLASS = "class";
+  static final String MAX_AHEAD_TIME_MS_CFG = "max-watermark-ahead-ms";
```

Comment:
Can you please add javadocs for these options? Also I think these constants should be public, so user can reuse them in case he's tweaking the config programatically.

Reply:
:+1: